File: src/napakalaki/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package napakalaki;

public class Treasure {
    private String name;
    private int goldCoins;
    private int minBonus;
    private int maxBonus;
    private TreasureKind type;

    public Treasure(String name, int goldCoins, int minBonus, int maxBonus, TreasureKind type) {
        this.name = name;
        this.goldCoins = goldCoins;
        this.minBonus = minBonus;
        this.maxBonus = maxBonus;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    
    public int getGoldCoins() {
        return goldCoins;
    }

    public void setGoldCoins(int goldCoins) {
        this.goldCoins = goldCoins;
    }
    
    public int getMinBonus() {
        return minBonus;
    }

    public void setMinBonus(int minBonus) {
        this.minBonus = minBonus;
    }

    public int getMaxBonus() {
        return maxBonus;
    }

    public void setMaxBonus(int maxBonus) {
        this.maxBonus = maxBonus;
    }

    public void setType(TreasureKind type) {
        this.type = type;
    }
    
    public TreasureKind getType() {
        return type;
    }

    public String toString() {
        return "name=" + name + ", goldCoins=" + goldCoins + ", minBonus=" + minBonus + ", maxBonus=" + maxBonus + ", type=" + type;
    }
    
    
}

File: P5Ruby/NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/treasure.rb
# encoding: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
# encoding: UTF-8
class Treasure
  attr_accessor :name,:gold_coins,:min_bonus,:max_bonus,:type
  def initialize(name,gold_coin, min, max, type)
    @name = name
    @min_bonus = min
    @max_bonus = max
    @type = type
    @gold_coins = gold_coin
  end
  
    
  def to_s
    "#{@name}, Oros #{@gold_coins}, tipo: #{@type} "
    
  end
  
end
